Modified AnonymiseFileNames

TheMasterKitty opened the pull request:
Added selection of file extensions for users to select (like ".png") so that those files can be anonymized by default while regular files stay under the same name.

(sorry for the confusion earlier, I'll try to get used to github)

TheMasterKitty committed:
Modified AnonymiseFileNames

Added selection of file extensions for users to select (like ".png") so that those files can be anonymized by default while regular files stay under the same name

redbaron2k7 commented on src/plugins/anonymiseFileNames/index.tsx:
```diff
@@ -112,15 +126,14 @@ export default definePlugin({
     anonymise(upload: AnonUpload) {
         if ((upload.anonymise ?? settings.store.anonymiseByDefault) === false) return upload.filename;
-
+        
```
You have an extra tab here

TheMasterKitty replied:
fixed in [19d4a5f](https://github.com/Vendicated/Vencord/pull/2797/commits/19d4a5f072e757ab253a20cda68a0d910a3d1e19)

redbaron2k7 commented on src/plugins/anonymiseFileNames/index.tsx:
```diff
@@ -112,15 +126,14 @@ export default definePlugin({
-        const ext = extIdx !== -1 ? file.slice(extIdx) : "";
-
+        const ext = extIdx !== -1 ? file.substring(extIdx) : "";
+        
```
Another extra tab

TheMasterKitty replied:
fixed in [19d4a5f](https://github.com/Vendicated/Vencord/pull/2797/commits/19d4a5f072e757ab253a20cda68a0d910a3d1e19)

redbaron2k7 commented on src/plugins/anonymiseFileNames/index.tsx:
```diff
@@ -42,6 +42,11 @@ const settings = definePluginSettings({
         type: OptionType.BOOLEAN,
         default: true,
     },
+    anonymiseByDefaultExts: {
+        description: "File extensions to anonymise names by default (list split by ', ')",
+        type: OptionType.STRING,
+        default: "",
+    },
```
Probably should add isValid logic here and change the wording to "(comma-separated list)"

TheMasterKitty replied:
what kind of isvalid logic r u looking for

redbaron2k7 replied:
In the settings to make sure the list of extensions is comma-separated/valid input.

TheMasterKitty replied:
k

redbaron2k7 replied:
If you want an example of what I mean I can show you on Discord

TheMasterKitty replied:
how would i do validation in settings?

redbaron2k7 commented on src/plugins/anonymiseFileNames/index.tsx:
```diff
@@ -67,7 +72,7 @@ const settings = definePluginSettings({
 export default definePlugin({
     name: "AnonymiseFileNames",
-    authors: [Devs.fawn],
+    authors: [ Devs.fawn, Devs.TheMasterKitty ],
```
```suggestion
    authors: [Devs.fawn, Devs.TheMasterKitty],
```

TheMasterKitty replied:
fixed

TheMasterKitty committed:
removed spaces

was most important bug

Co-authored-by: Aidan <76609332+redbaron2k7@users.noreply.github.com>

TheMasterKitty committed:
fixed major vulnerabilties and reworded setting

removed extra tabs on line 131 and 136
reworded anonymiseByDefaultExts setting and added default for example usage

TheMasterKitty commented:
dont mind me am a lil silly :3

TheMasterKitty committed:
Added Validation

Code by redbaron2k7

TheMasterKitty committed:
fixed validator function

had errors

TheMasterKitty committed:
Merge branch 'main' into AnonymiseFileNames

TheMasterKitty committed:
Merge branch 'Vendicated:main' into AnonymiseFileNames
